lstests/catchup_test.py

```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env python3
+
+import time
+
+from core.logger import ScreenWriter, FileWriter, Logger
+from core.service import Service, Cluster, LauncherServiceError, BlockchainError, SyncError
+
+TEST_PLUGIN = "eosio::txn_test_gen_plugin"
+CREATE_URL = "/v1/txn_test_gen/create_test_accounts"
+START_URL = "/v1/txn_test_gen/start_generation"
+STOP_URL = "/v1/txn_test_gen/stop_generation"
+PRIVATE_KEY = "5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3"
+CREATE_STR = f"[\"eosio\", \"{PRIVATE_KEY}\"]"
+START_STR = "[\"salt\",10,10]"
+CATCHUP_ROUNDS = 3
+
+
+def init_cluster():
+    test = "catchup"
+    logger = Logger(ScreenWriter(threshold="info"),
+                    FileWriter(filename=f"{test}-info.log", threshold="info", monochrome=True),
+                    FileWriter(filename=f"{test}-debug.log", threshold="debug", monochrome=True),
+                    FileWriter(filename=f"{test}-trace.log", threshold="trace", monochrome=True))
+    service = Service(logger=logger)
+    cluster = Cluster(service=service, node_count=3, pnode_count=3, producer_count=3,
+                      dont_newacco=True, extra_configs=[f"plugin={TEST_PLUGIN}"])
+    return cluster
+
+
+def create_accounts(clus):
+    clus.info(">>> [Catch-up Test] Create Test Accounts")
+    for _ in range(5):
+        cx = clus.call("send_raw", url=CREATE_URL, string_data=CREATE_STR, node_id=1,
+                        retry=0, level="trace", error_level="trace", dont_raise=True)
+        if '"name": "account_name_exists_exception"' in cx.response_text:
+            return clus.check_sync().block_num
+        time.sleep(1)
+    raise BlockchainError(f"Failed to create test accounts with {TEST_PLUGIN}")
+
+
+def start_gen(clus, begin):
+    clus.info(">>> [Catch-up Test] Generate Transactions")
+    clus.call("send_raw", url=START_URL, string_data=START_STR, node_id=1, level="trace")
+    blks = 30
+    clus.info(f"Generation begins at block num {begin}")
+    clus.debug(f"Wait {blks * 0.5} seconds for {blks} blocks...")
+    time.sleep(blks * 0.5)
+    end = clus.check_sync().block_num
+    if end - begin < blks:
+        raise BlockchainError(f"Blocks produced ({end - begin}) are fewer than expected ({blks})")
+    clus.info(f"Generation ends at block num {end}")
+    return end
+
+
+def count_gen(clus, begin, end):
```

Comment:
You can use similar logic for verifying average transactions received (see change request above)

Reply:
added check for average transactions per block, please check.